pkg/engine/workflow/engine.go

```diff
@@ -166,13 +176,14 @@ func Execute(pl *ActivePlan, em *engine.Metadata, c client.Client, enh renderer.
 				// stopped in the spirit of "fail-loud-and-proud".
 				switch {
 				case errors.Is(err, engine.ErrFatalExecution):
-					phaseStatus.Status = v1beta1.ExecutionFatalError
-					stepStatus.Status = v1beta1.ExecutionFatalError
-					planStatus.Status = v1beta1.ExecutionFatalError
+					phaseStatus.Set(v1beta1.ExecutionFatalError)
```

Comment:
it would be nice to move some of this out to a further helper function that sets all 3 at once, this is repetitive through the cases and is distracting me :joy:

Reply:
:D I see, sometimes we set three, sometimes we set two, sometimes one... and it's not a doubly linked list so you cannot traverse from step up to propagate it :/ do you feel like